py/visdom/server.py

```diff
@@ -319,6 +322,36 @@ def load_state(self):
 
         return state
 
+    def load_user_settings(self):
+        settings = {}
+
+        """Determines & uses the platform-specific root directory for user configurations."""
+        if platform.system() == "Windows":
+            base_dir = os.getenv('APPDATA')
+        elif platform.system() == "Darwin": # osx
+            base_dir = os.path.expanduser('~/Library/Preferences')
+        else:
+            base_dir = os.getenv('XDG_CONFIG_HOME', os.path.expanduser('~/.config'))
+        config_dir = os.path.join(base_dir, "visdom")
+
+        # initialize user style
+        user_css = ""
+        logging.error("initializing")
+        home_style_path = os.path.join(config_dir, "style.css")
+        if os.path.exists(home_style_path):
+            with open(home_style_path, "r") as f:
+                user_css += "\n" + f.read()
+        project_style_path = os.path.join(self.env_path, "style.css")
+        if os.path.exists(project_style_path):
+            with open(project_style_path, "r") as f:
+                user_css += "\n" + f.read()
```

Comment:
Would it ever be the case where we'd want to combine both of these styles rather than directly overwrite? Not necessarily that we need to support this, but proposing the thought. If we decide against it you should include a note about this behavior in the docs.

Reply:
Of course, both options (extending and replacing style-file) don't exclude each other. However, I am unsure, if this makes the feature too complicated.

Let's run through the idea:

-----

In case of both features, I think, the best search order for the style to serve is:
1. replacing style-file for current env-path
2. replacing style-file for user
3. default style-file with extending style-file, first, by extending using the user style-file, then by extending using the env-path style file.
   (Currently, this PR implements only this option.)

What would be the naming scheme?
1. `style.css` for replacing style-files
2. `extend.css` for extending style-files

----

What do you think?

Reply:
Ah thinking about this more with your context, your version is already best inline with CSS standards - repeated rules already take the later declaration, so you're set up in a case where overriding will occur in the correct direction here. If a user really wants to completely override they're free to do so. Thanks for enumerating here though!